### ISSUE TITLE
Wanted: Ability to configure splunkforwarding to non-chef managed server

### ISSUE THREAD

**Issue by ampledata** [chef-cookbooks/chef-splunk#15]
Currently the 'client' Recipe searches for chef-managed servers to which to forward. Instead I'd like to forward to Splunk Cloud servers, which would probably be defined by a node attrib, or data bag.

**Comment by trumant** [comment]
I'll be happy to take a swing at this.

**Comment by trumant** [comment]
@ampledata , @jtimberman and @hrez - Would appreciate your feedback on initial work: https://github.com/trumant/chef-splunk/commit/251072ce66aa3c39ba1dc9972c594f6c534598fe

Still working on getting test kitchen tests rounded out to prove that data bag and node attribute configuration works.

**Comment by thomasbiddle** [comment]
@jtimberman @trumant Any update on implementing this? +1 on it being hugely beneficial.

**Comment by trumant** [comment]
@thomasbiddle - I didn't end up taking it any further because I ended up replacing my usage of this cookbook with an Ansible role as part of a broader move to Ansible.

**Comment by Sauraus** [comment]
There is a simple way of doing this, add an attribute and if the server search returns nil, set the servers based on the values in the attribute.

**Comment by akemner** [comment]
+1. this would be beneficial for people using 'Splunk Cloud'

**Comment by sworisbreathing** [comment]
This could also be done by defining the outputs.conf in an app bundle and then using the `splunk_app` resource.

**Comment by tas50** [comment]
#54 has a solution to this, but we need additional testing before it can be merged.

**Comment by borgified** [comment]
my workaround to this was to set

```
"splunk": {
  "outputs_conf": {
    "server": "1.1.1.1""
  }
}
```

the resulting output.conf file will end up looking like this: 
```
[tcpout]
defaultGroup = splunk_indexers_9997
disabled=false

[tcpout:splunk_indexers_9997]
server=
forwardedindex.0.whitelist = .*
forwardedindex.1.blacklist = _.*
forwardedindex.2.whitelist = _audit
forwardedindex.filter.disable = false
server = 1.1.1.1
```

and that seems to work for the splunkforwarder

**Comment by rarsan** [comment]
Fixed with #93 . Closing.
